### PR TITLE
Pic filename

### PR DESCRIPTION
**Pull request by i92guboj**
This would convert pic_id to pic_filename, and add the necessary code to handle the fact that this field, from now on, will hold the full name (.ext included). Where necessary, conditional branching to detect legacy file names has been added, and a couple of silent update points have also been introduced, so, the existing installs should be automatically fixed.

Of course, this needs testing, and you should backup if you really want to test this on a production database.

It shouldn't do much harm, but still...

So, to sum up:

- field pic_id was renamed to pic_filename
- data type was changed from int to varchar(255)
- database/3.0.2_to_3.1.sql script was added to ALTER the items table
- import_items.csv was modified to add an example of the newly available field
- php was modified to be in agreement with the changes above, and also to manage the migration of already existing elements in the altered field

### CODE REVIEW

**Comment by i92guboj** [comment]
Hi, as per the conversation in #1026 I am proposing this patch.

Also, please, note that I am a rookie to all this nodejs stuff. Bower touched several files, and git has commited that, I just let it do its work. While I know how to work with git, I have no idea what bower is so, please, forgive me if something useless went through the commit process.

File: application/helpers/table_helper.php [hunk 1]
```diff
@@ -268,13 +268,21 @@ function get_item_data_row($item, $controller)
...
+			$images = glob('./uploads/item_pics/' . $item->pic_filename . '.*');
+		}
+		else {
+			// preferred
+			$images = glob('./uploads/item_pics/' . $item->pic_filename);
+		}
 		if (sizeof($images) > 0)
 		{
-			$image .= '<a class="rollover" href="'. base_url($images[0]) .'"><img src="'.site_url('items/pic_thumb/'.$item->pic_id).'"></a>';
+			$image .= '<a class="rollover" href="'. base_url($images[0]) .'"><img src="'.site_url('items/pic_thumb/' . pathinfo($item->pic_filename, PATHINFO_BASENAME)) . '"></a>';
```
Comment (jekkos) [inline application/helpers/table_helper.php]: In fact this function call will always fallback to the filename without extension, if I read this right? So in this case even if the database column has the file extension, we will always be using the legacy glob functionality inside pic_thumb

File: application/helpers/table_helper.php [hunk 2]
```diff
@@ -268,13 +268,21 @@ function get_item_data_row($item, $controller)
...
-		$images = glob('./uploads/item_pics/' . $item->pic_id . '.*');
+		$ext = pathinfo($item->pic_filename, PATHINFO_EXTENSION);
+		if($ext == '')
+		{
+			// legacy
+			$images = glob('./uploads/item_pics/' . $item->pic_filename . '.*');
+		}
+		else {
+			// preferred
+			$images = glob('./uploads/item_pics/' . $item->pic_filename);
```
Comment (i92guboj) [inline application/helpers/table_helper.php]: As said, I am no php master. But, as I see it, glob() still serves as a way not to reference non-existent files. If the pattern doesn't match a file, it will return an empty array, I am not sure if for php trying to open a non-existant file can have any negative implication.

If you think glob() is superfluous here, I have of course remove it and assume the file exists?

File: application/helpers/table_helper.php (hunk 1 above)
Comment (i92guboj) [inline application/helpers/table_helper.php]: No, you have a point here, but I think it's actually the other way around. AFAIK, PATHINFO_BASENAME is "filename.ext", which is different from PATHINFO_FILENAME, which would hold just "filename" (no .ext). Just like the *nix basename command line tool.

So, pic_thumb() is always called with full filename, and probably the code inside pic_thumb() to handle the "no-extension case" is redundant. So, probably, just having it in one place or the other should be enough. 

I have to double check this, right now I am busy. I think I'll get to that this evening.
ank you for taking the time to review this, it's much appreciated.

File: application/helpers/table_helper.php (hunk 2 above)
Comment (i92guboj) [inline application/helpers/table_helper.php]: Ok, I've been looking into this. The code in pic_thumb() is useless, as you noticed.

The code in the table helper, needs one modification. Using pathinfo on $item->pic_filename to construct the url is incorrect, since it can be with or without extension. You need to use $images[0], and then PATHINFO_BASENAME takes meaning. Maybe that's what drove you to confusion.

**Comment by i92guboj** [comment]
The last commit simplifies the code a bit and corrects a little bug. pic_thumb() now receives "filename.ext" (the whole basename) as an argument. 

It's unclear to me whether using glob() when the file name is "full" is really necessary or has any advantage (regarding security or whatever) over using the string directly and hoping that the file is there (if it isn't then the server should issue 404, shouldn't it?).

**Comment by i92guboj** [comment]
For example, in pic_thumb() we have
`		$images = glob('./uploads/item_pics/' . $pic_filename);`
and below we have
`		if(sizeof($images) > 0)`
so this is a way (smart or not, I have no idea) to control whether the file exists. 

So, it glob() here has a purpose. Whether that purpose is legit or not, that's open for discussion. I guess there are better ways to infer if a given file exist, that are more legible. If you think it's worth the effort, I can certainly take care of that.

**Comment by daN4cat** [comment]
Thank you for the contribution, I will try to review the code past Christmas together with other contributions.

**Comment by daN4cat** [comment]
@i92guboj could you please rebase this PR with the latest from master? I also see that @jekkos requested changes, have you addressed those?

Thanks for the work by the way.

**Comment by i92guboj** [comment]
Hi!

The requested changes are adressed as far as I can remember. But, as a rebase is due, I'll review everything ASAP. Can't promise anything but I should be able to do so in the next couple of days if nothing gets in the middle.

Thanks!

**Comment by daN4cat** [comment]
@i92guboj any update on the matter?

**Comment by i92guboj** [comment]
Hi and sorry for the delay. This should be fine now.

**Comment by daN4cat** [comment]
@i92guboj sorry one more rebase due to the last commit to master. Thanks

**Comment by i92guboj** [comment]
That should do it :)

**Comment by hrshadhin** [comment]
Hi, I'm using OSPOS latest version and have a issue with item thumb image.
`/items/pic_thumb/Recipe-Chicken.png` 
above url not map on nginx. its show 404 error.
